packages/syntax/tests/datetime.test.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  isValidDatetime,
+  ensureValidDatetime,
+  normalizeDatetime,
+  normalizeDatetimeAlways,
+  InvalidDatetimeError,
+} from '../src'
+import * as readline from 'readline'
+import * as fs from 'fs'
+
+describe('datetime validation', () => {
+  const expectValid = (h: string) => {
+    ensureValidDatetime(h)
+    normalizeDatetime(h)
+    normalizeDatetimeAlways(h)
+  }
+  const expectInvalid = (h: string) => {
+    expect(() => ensureValidDatetime(h)).toThrow(InvalidDatetimeError)
+  }
+
+  it('conforms to interop valid datetimes', () => {
+    const lineReader = readline.createInterface({
+      input: fs.createReadStream(
+        `${__dirname}/interop-files/datetime_syntax_valid.txt`,
+      ),
+      terminal: false,
+    })
+    lineReader.on('line', (line) => {
+      if (line.startsWith('#') || line.length == 0) {
+        return
+      }
+      if (!isValidDatetime(line)) {
+        console.log(line)
+      }
+      expectValid(line)
+    })
+  })
+
+  it('conforms to interop invalid datetimes', () => {
+    const lineReader = readline.createInterface({
+      input: fs.createReadStream(
+        `${__dirname}/interop-files/datetime_syntax_invalid.txt`,
+      ),
+      terminal: false,
+    })
+    lineReader.on('line', (line) => {
+      if (line.startsWith('#') || line.length == 0) {
+        return
+      }
+      expectInvalid(line)
+    })
+  })
+
+  it('conforms to interop invalid parse (semantics) datetimes', () => {
+    const lineReader = readline.createInterface({
+      input: fs.createReadStream(
+        `${__dirname}/interop-files/datetime_parse_invalid.txt`,
+      ),
+      terminal: false,
+    })
+    lineReader.on('line', (line) => {
+      if (line.startsWith('#') || line.length == 0) {
+        return
+      }
+      expectInvalid(line)
+    })
+  })
+})
+
+describe('normalization', () => {
+  it('normalizes datetimes', () => {
```

Comment:
I believe this fun vector originally seen from retr0.id still makes it through, where the normalized date ends-up before year zero, which is invalid:
```js
> normalizeDatetime('0000-01-01T00:00:00+01:00')
'-000001-12-31T23:00:00.000Z'
```

In `toSimplifiedISOSafe()` you'll find an extra validation check on the normalized output, which is what currently catches this.

Reply:
just added a test for that, lmk how it looks 👍